### PR TITLE
feat(#451): Create cloudfront invalidator

### PR DESCRIPTION
**Pull request by ChewingGlass**

### CODE REVIEW

File: packages/entity-invalidator/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+      lastId = entities[entities.length - 1].id;
+      totalProgress += entities.length;
+      console.log(`Processed ${totalProgress} / ${totalCount}`);
+    }
+  } while (entities.length === limit);
+}
+
+function getPaths(entity: KeyToAsset): string[] {
+  const v1 = `v1/${entity.address}`;
```
Comment (deasydoesit) [inline packages/entity-invalidator/src/index.ts:107]: This needs to start with a leading slash

File: packages/entity-invalidator/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+  if ((entity.entityKeyStr?.length || 0) >= 200) {
+    return [v1];
+  }
+
+  return [entity.entityKeyStr!, v1];
```
Comment (deasydoesit) [inline packages/entity-invalidator/src/index.ts:112]: Same here

**Review by deasydoesit — APPROVED** [review]

File: packages/entity-invalidator/package.json [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    "@types/deep-equal": "^1.0.1",
+    "@types/node": "^18.11.11",
+    "@types/pg": "^8.6.6",
+    "@types/yargs": "^17.0.24",
+    "git-format-staged": "^2.1.3",
+    "ts-loader": "^9.2.3",
+    "ts-node": "^10.9.1",
+    "ts-node-dev": "^2.0.0",
+    "typescript": "^4.8.4",
+    "yarn": "^1.22.18"
```
Comment (bryzettler) [inline packages/entity-invalidator/package.json:57]: rm yarn

File: packages/entity-invalidator/package.json [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    "@types/bn.js": "^5.1.1",
+    "@types/deep-equal": "^1.0.1",
+    "@types/node": "^18.11.11",
+    "@types/pg": "^8.6.6",
+    "@types/yargs": "^17.0.24",
+    "git-format-staged": "^2.1.3",
+    "ts-loader": "^9.2.3",
+    "ts-node": "^10.9.1",
+    "ts-node-dev": "^2.0.0",
+    "typescript": "^4.8.4",
```
Comment (bryzettler) [inline packages/entity-invalidator/package.json:56]: ^5.2.2

File: packages/entity-invalidator/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,30 @@
+# Specify the base image
+FROM node:16-alpine AS BUILD_IMAGE
```
Comment (bryzettler) [inline packages/entity-invalidator/Dockerfile:2]: node:18-alpine

File: packages/entity-invalidator/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+COPY src src
+COPY tsconfig.build.json tsconfig.json
+
+RUN yarn global add typescript
+RUN yarn run build
+
+RUN npm prune --production
+
+FROM node:16-alpine
```
Comment (bryzettler) [inline packages/entity-invalidator/Dockerfile:18]: node:18-alppine

File: packages/entity-invalidator/Dockerfile [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+WORKDIR /usr/src/app
+
+COPY package.json ./
+
+RUN yarn install
+
+COPY src src
+COPY tsconfig.build.json tsconfig.json
+
+RUN yarn global add typescript
```
Comment (bryzettler) [inline packages/entity-invalidator/Dockerfile:13]: rm, no longer needed

File: packages/entity-invalidator/Dockerfile [hunk 8]
```diff
@@ -0,0 +1,30 @@
+# Specify the base image
+FROM node:16-alpine AS BUILD_IMAGE
+
+WORKDIR /usr/src/app
+
```
Comment (bryzettler) [inline packages/entity-invalidator/Dockerfile:5]: eye a new dockerfile
```COPY yarn.deploy.lock ./
ENV YARN_LOCKFILE_FILENAME=yarn.deploy.lock

RUN corepack enable
RUN yarn set version stable
RUN yarn install```

**Pull request merged**